File: blender/fontemon_blender_addon/GameToFont/verticalLineToCommand.py
class VerticalLineToPair:
    def __init__(self, dy: float, thenDx: float) -> None:
        self.dy = dy
        self.thenDx = thenDx


def verticalLineToCommand(pairs, lastDy):
    # type: (list[VerticalLineToPair], float) -> str
    return f"{' '.join([f'{p.dy} {p.thenDx}' for p in pairs])} {lastDy} vlineto\n"

File: imageToCharStringConverter/gui/src/verticalLineToCommand.ts
/**
 * Note: There are two forms to this command,
 * this uses the odd numbered form.
 * @odd numbered form
 */
export const verticalLineToCommand = ({
  pairs,
  lastDy,
}: {
  pairs: VerticalLineToPair[];
  lastDy: number;
}) =>
  `${pairs
    .map(({ dy, thenDx }) => `${dy} ${thenDx}`)
    .join(" ")} ${lastDy} vlineto\n`;

/**
 * Input pair to the vertical Line To Command.
 * The dy is applied before the dx,
 * that is why it is called dy and thenDx
 */
export interface VerticalLineToPair {
  readonly dy: number;
  readonly thenDx: number;
}
